apps/scan/backend/src/app_flow.ts

```diff
@@ -0,0 +1,119 @@
+import { InsertedSmartCardAuthApi } from '@votingworks/auth';
+import { doesUsbDriveRequireCastVoteRecordSync } from '@votingworks/backend';
+import { assert } from '@votingworks/basics';
+import { UsbDrive } from '@votingworks/usb-drive';
+import {
+  isElectionManagerAuth,
+  isPollWorkerAuth,
+  isSystemAdministratorAuth,
+} from '@votingworks/utils';
+import { BALLOT_BAG_CAPACITY } from './globals';
+import { Store } from './store';
+import { constructAuthMachineState } from './util/construct_auth_machine_state';
+
+/**
+ * Determines whether the VxScan is ready to scan. There are circumstances when
+ * there is a ballot inserted into the machine, but VxScan is not able to accept
+ * it. For example, when the ballot bag is full, or when a user has logged in
+ * with a smartcard.
+ */
+export async function isReadyToScan({
+  auth,
+  store,
+  usbDrive,
+}: {
+  auth: InsertedSmartCardAuthApi;
+  store: Store;
+  usbDrive: UsbDrive;
+}): Promise<boolean> {
+  const authStatus = await auth.getAuthStatus(constructAuthMachineState(store));
+  if (
+    authStatus.status === 'logged_out' &&
+    authStatus.reason === 'no_card_reader'
+  ) {
+    return false;
+  }
+
+  const electionDefinition = store.getElectionDefinition();
+
+  if (
+    !electionDefinition &&
+    authStatus.status === 'logged_out' &&
+    authStatus.reason === 'no_card'
+  ) {
+    return false;
+  }
+
+  if (
+    authStatus.status === 'logged_out' &&
+    authStatus.reason === 'card_error'
+  ) {
+    return false;
+  }
+
+  if (authStatus.status === 'logged_out' && authStatus.reason !== 'no_card') {
+    return false;
+  }
+
+  if (
+    authStatus.status === 'checking_pin' &&
+    authStatus.user.role === 'system_administrator'
+  ) {
+    return false;
+  }
+
+  if (isSystemAdministratorAuth(authStatus)) {
+    return false;
+  }
+
+  if (authStatus.status === 'checking_pin') {
+    return false;
+  }
+
+  if (!electionDefinition) {
+    return false;
+  }
+
+  if (isElectionManagerAuth(authStatus)) {
+    return false;
+  }
```

Comment:
Could we consolidate some of these? It feels like these checks are leftover from when we had specific screens for each reason scanning wasn't ready. Now that it's just a boolean I think we could reduce the number of checks. eg the checks for `!isSystemAdministratorAuth`, `!isPollWorkerAuth`, and `!isElectionManagerAuth` seem redundant with L107 

```
assert(authStatus.status === 'logged_out' && authStatus.reason === 'no_card');
```

and the various logged out checks are redundant with [L54](https://github.com/votingworks/vxsuite/pull/4403/files#diff-d209b5ccb87bea87d1299afbc7f9559a495ae9f5247d16380b060d83ab5d1f8fR54-R56)

(On the other hand it may make sense to keep these very specific checks if we plan to eventually move some decisions about what to render to the backend)

Reply:
Though I think it may be a worthwhile refactor, we're unlikely to do it in the near future. So I agree that simplifying it is best for now.